src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -297,6 +300,31 @@ uploadContainerScan apiOpts ProjectRevision{..} metadata scan = fossaReq $ do
   resp <- req POST (containerUploadUrl baseUrl) (ReqBodyJson scan) jsonResponse (baseOpts <> opts)
   pure $ responseBody resp
 
+uploadNativeContainerScan ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  ApiOpts ->
+  ProjectRevision ->
+  ProjectMetadata ->
+  NativeContainer.ContainerScan ->
+  m UploadResponse
+uploadNativeContainerScan apiOpts ProjectRevision{..} metadata scan = fossaReq $ do
+  supportsNativeScan <- orgSupportsNativeContainerScan <$> getOrganization apiOpts
+  -- Sanity Check!
+  if not supportsNativeScan
```

Comment:
[Optional] This check seems redundant because `uploadScan` already does it. Is this a better place to gate? Or would it be reasonable to expect the caller of this function to do the gating?

Reply:
Yeah - I added this guard in `uploadScan` so I can have a unit test. It is hard to mock failure in `req` with MockApi. So I went with simpler option.

Reply:
It's not important, but I think it's totally reasonable to have the check only be in `uploadScan` if you wanted it to be. The gating should apply to both the mock and the real implementation, correct?